components/brave_wallet_ui/components/extension/assets-panel/index.tsx

```diff
@@ -138,25 +142,41 @@ const AssetsPanel = (props: Props) => {
   )
 
   const tokenPriceIds = React.useMemo(() =>
-    userTokens && balances
+    selectedAccount && userTokens && tokenBalancesRegistry
       ? userTokens
-          .filter(token => new Amount(balances[token.contractAddress]).gt(0))
+          .filter(token => new Amount(
+            getBalance(selectedAccount, token, tokenBalancesRegistry)).gt(0))
           .filter(token => !token.isErc721 && !token.isErc1155 && !token.isNft)
           .map(getPriceIdForToken)
       : [],
-    [userTokens, balances]
+    [userTokens, tokenBalancesRegistry]
   )
 
   const {
     data: spotPriceRegistry,
     isLoading: isLoadingSpotPrices
   } = useGetTokenSpotPricesQuery(
-    tokenPriceIds.length && !isLoadingBalances && !isFetchingBalances
-      ? { ids: tokenPriceIds }
-      : skipToken,
+    tokenPriceIds.length &&
+      !isLoadingBalances && !isFetchingBalances && defaultFiatCurrency
+        ? { ids: tokenPriceIds, toCurrency: defaultFiatCurrency }
+        : skipToken,
     querySubscriptionOptions60s
   )
 
+  const areTokenBalancesLoaded = React.useMemo(() => {
+    return (
+      selectedAccount &&
+      tokenBalancesRegistry &&
+      !isLoadingBalances &&
+      !isFetchingBalances
+    )
+  }, [
+    selectedAccount,
+    tokenBalancesRegistry,
+    isLoadingBalances,
+    isFetchingBalances
+  ])
```

Comment:
perf: probably will run faster if we compute this on render instead of memoizing

Reply:
Fixed here: [313064f](https://github.com/brave/brave-core/pull/19337/commits/313064f7fb0f6747da8d04405c5c4859d59c3433)